### ISSUE TITLE
Add support for symlink following for regular files/unix domain sockets (in chroot/efs )

### ISSUE THREAD

**Issue by NirjharRoyiitk** [gramineproject/gramine#1193]
### Description of the feature

Add support for symlink following for regular files/unix domain sockets (in chroot/efs ). This will enable applications to use system calls like ```symlink```, ```symlinkat```, ```link```, ```linkat```. I will be working on this feature. 

### Why Gramine should implement it?

Many applications use hard and soft links.

**Comment by NirjharRoyiitk** [comment]
I will be working on this feature.

**Comment by dimakuv** [comment]
> chroot/efs

What's that? Is `e` just a typo?

> I will be working on this feature.

Before starting the work on this feature, please present your proposed design to the Gramine team. Otherwise it may be a waste of time, since Gramine maintainers may not approve the design itself.

**Comment by NirjharRoyiitk** [comment]
@dimakuv Oh I meant chroot and Encrpypted File System (EFS).

**Comment by dimakuv** [comment]
Ok, got it. Please don't limit yourself to `chroot` and `encrypted` only, please also add it to `tmpfs`. This way you'll cover all three currently available FS types (note that pseudo-files are already covered, see the use of `follow_link()` callback).

And just to reiterate: **Before starting the work on this feature, please present your proposed design to the Gramine team.**
